applications/crossbar/src/modules/cb_cdrs.erl

```diff
@@ -255,465 +231,172 @@ validate(Context, CDRId) ->
 %% @doc
+%% Generate specific view options for the path.
 %% @end
 %%--------------------------------------------------------------------
--type view_option_fun() :: fun((api_binary(), cb_context:context(), gregorian_seconds(), gregorian_seconds()) -> {'ok', crossbar_doc:view_options()}).
-
--spec create_view_options(api_binary(), cb_context:context()) ->
-                                 {'ok', crossbar_doc:view_options()} |
-                                 cb_context:context().
--spec create_view_options(api_binary(), view_option_fun(), cb_context:context()) ->
-                                 {'ok', crossbar_doc:view_options()} |
-                                 cb_context:context().
-create_view_options(OwnerId, Context) ->
-    create_view_options(OwnerId, fun create_view_options/4, Context).
-
-create_view_options(OwnerId, Fun, Context) ->
-    case cb_modules_util:range_view_options(Context) of
-        {CreatedFrom, CreatedTo} ->
-            Fun(OwnerId, Context, CreatedFrom, CreatedTo);
-        Context1 -> Context1
-    end.
-
--spec create_view_options(api_binary(), cb_context:context(), gregorian_seconds(), gregorian_seconds()) ->
-                                 {'ok', crossbar_doc:view_options()}.
-create_view_options('undefined', Context, CreatedFrom, CreatedTo) ->
-    StartKey = maybe_get_start_key_from_req(Context, CreatedTo),
-    {'ok'
-    ,props:filter_undefined(
-       [{'startkey', StartKey}
-       ,{'endkey', CreatedFrom}
-       ,{'limit', pagination_page_size(Context)}
-       ,'descending'
-       ])
+-spec get_view_options(req_nouns()) -> {api_ne_binary(), crossbar_view:options()}.
+get_view_options([{<<"cdrs">>, []}, {?KZ_ACCOUNTS_DB, _}|_]) ->
+    {?CB_LIST, []};
+get_view_options([{<<"cdrs">>, []}, {<<"users">>, [OwnerId]}|_]) ->
+    {?CB_LIST_BY_USER
+    ,[{'start_key_map', [OwnerId]}
+     ,{'end_key_map', [OwnerId]}
+     ]
     };
-create_view_options(OwnerId, Context, CreatedFrom, CreatedTo) ->
-    StartKey = maybe_get_start_key_from_req(Context, [OwnerId, CreatedTo]),
-    {'ok'
-    ,props:filter_undefined(
-       [{'startkey', StartKey}
-       ,{'endkey', [OwnerId, CreatedFrom]}
-       ,{'limit', pagination_page_size(Context)}
-       ,'descending'
-       ])
-    }.
-
--spec create_interaction_view_options(api_binary(), cb_context:context(), pos_integer(), pos_integer()) ->
-                                             {'ok', crossbar_doc:view_options()}.
-create_interaction_view_options('undefined', Context, CreatedFrom, CreatedTo) ->
-    StartKey = maybe_get_start_key_from_req(Context, [CreatedTo]),
-    {'ok'
+get_view_options([{<<"cdrs">>, [?PATH_INTERACTION]}, {?KZ_ACCOUNTS_DB, _}|_]) ->
+    {?CB_INTERACTION_LIST
     ,props:filter_undefined(
-       [{'startkey', StartKey}
-       ,{'endkey', [CreatedFrom, kz_json:new()]}
-       ,{'limit', pagination_page_size(Context)}
+       [{'start_key_map', []}
+       ,{'end_key_map', crossbar_view:suffix_key_fun([kz_json:new()])}
```

Comment:
`[kz_json:new()]` is not covered by the `crossbar_view:keymap()` type.

Reply:
added to `crossbar_view:keymap()`

Reply:
Not quite :) Empty json object was added but this is passing in a list of json objects. Should be `kz_json:objects()` instead of `?EMPTY_JSON_OBJECT`.